R/import_metadata_and_counts.R

```diff
@@ -0,0 +1,118 @@
+#' Checks importing criteria for new atlas 
+#' Generats counts per million from provided raw counts
+#'
+#' @param sce_obj A SingleCellExperiment object from RDS
+#' @param cache_dir Optional character vector of length 1. A file path on
+#'   your local system to a directory (not a file) that will be used to store
+#'   `metadata.parquet`
+#' @export
+#' @return A metadata.parquet strip from SCE object. 
+#' Directories store counts and counts per million in the provided cache directory.
+#' @importFrom assertthat assert_that
+#' @importFrom checkmate check_tibble check_directory_exists check_set_equal check_true check_character check_subset check_file_exists
+#' @importFrom dplyr tbl select
+#' @importFrom cli cli_alert_info
+#' @importFrom glue glue
+#' @importFrom rlang .data
+#' @examples
+#' data(sample_sce_obj)
+#' import_metadata_counts(sample_sce_obj,
+#'                        cache_dir = get_default_cache_dir())
+import_metadata_counts <- function(
+  sce_obj,  
+  cache_dir = get_default_cache_dir()) {
+  original_dir <- file.path(cache_dir, "original")
+  
+  # Identify metadata and counts matrix
+  metadata_tbl <- sce_obj@metadata$data
+  counts_matrix <- sce_obj@assays@data$X
+  
+  # Convert to tibble if metadata_tbl is not a tibble
+  metadata_tbl <- metadata_tbl |> as_tibble()
+  
+  # Create file_id_db from dataset_id
+  metadata_tbl <-
+    metadata_tbl |> mutate(file_id_db = .data$dataset_id |> openssl::md5() |> as.character())
+  sce_obj@metadata$data <- metadata_tbl
+  
+  # Remove existing reducedDim slot to enable get_SCE API functionality 
+  if (length(names(SingleCellExperiment::reducedDims(sce_obj))) >0 ) {
+    SingleCellExperiment::reducedDims(sce_obj) <- NULL
+  }
+  
+  # create original and cpm folders in cache directory if not exist (in order to append new counts to existing ones)
+  if (!dir.exists(original_dir)) {
+    dir.create(cache_dir, "original", recursive = TRUE)
```

Comment:
I don't think think this function actually takes multiple path arguments. 
```suggestion
    cache_dir |> file.path("original") |> dir.create(recursive = TRUE)
```